src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -161,3 +176,144 @@ export const handleRedirect: RequestHandler<
     redirectPath,
   })
 }
+
+/**
+ * Handler for GET /:formId/publicform endpoint
+ * @returns 200 if the form exists
+ * @returns 404 if form with formId does not exist or is private
+ * @returns 410 if form has been archived
+ * @returns 500 if database error occurs or if the type of error is unknown
+ */
+export const handleGetPublicForm: RequestHandler<{ formId: string }> = async (
+  req,
+  res,
+) => {
+  const { formId } = req.params
+
+  const formResult = await getFormIfPublic(formId).andThen((form) =>
+    FormService.checkFormSubmissionLimitAndDeactivateForm(form),
+  )
+
+  // Early return if form is not public or any error occurred.
+  if (formResult.isErr()) {
+    const { error } = formResult
+    logger.error({
+      message: 'Error retrieving public form',
+      meta: {
+        action: 'handleGetPublicForm',
+        ...createReqMeta(req),
+        formId,
+      },
+      error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(error)
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+
+  const form = formResult.value
+  const publicFormView = form.getPublicView()
+  const { authType } = form
+
+  // Shows the client a form based on how they are authorized
+  // If the client is MyInfo, we have to prefill the form
+  switch (authType) {
+    case AuthType.SP:
+    case AuthType.CP: {
+      // Form is valid, check for SPCP/MyInfo data.
+      return SpcpFactory.getSpcpSession(authType, req.cookies)
+        .map(({ userName }) =>
+          res.json({
+            form: publicFormView,
+            spcpSession: { userName },
+          }),
+        )
+        .mapErr((error) => {
+          logger.error({
+            message: 'Error getting public form',
```

Comment:
Only log if the error type is not MissingJwtError? If not our logs will get spammed when ppl visit out forms without logging into spcp

Reply:
good point, added a more descriptive comment to ensure this knowledge is transmitted